### PR TITLE
Add new bytewax feedstock

### PR DESCRIPTION
**Pull request by cav71**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

@conda-forge/staged-recipes, ready for review!

### CODE REVIEW

**Comment by zklaus** [comment]
@cav71, are you happy with the PR? Once you think you are ready, you will have to tag the team, i.e. write a new comment here that mentions @<the right team> (conda-forge/staged-recipes?). This will spring the bot into action, assign the right labels and get the right people's attention. Alas, I don't have permissions and thus am not one of them.

File: recipes/bytewax-feedstock/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/bytewax/bytewax/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 993fe70f3a206adc52aae1099c00e14a92ec55387dda4b0cda2b6f6fbf0ed35b
+  patches:
+    - ssl.patch
```
Comment (ocefpaf) [inline recipes/bytewax-feedstock/meta.yaml:12]: Can you add a comment here explaining the patch and/or if it is upstream?

File: recipes/bytewax-feedstock/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+requirements:
+  build:
+    - git
+    - curl
+    - {{ compiler('cxx') }}
+    - make
+    - pkg-config
+    - python
+
```
Comment (ocefpaf) [inline recipes/bytewax-feedstock/meta.yaml:27]:
```suggestion
```

File: recipes/bytewax-feedstock/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - ssl.patch
+
+build:
+  skip: True  # [win]
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - git
```
Comment (ocefpaf) [inline recipes/bytewax-feedstock/meta.yaml:21]: Why do you need git in build?

```suggestion
    - git
```

File: recipes/bytewax-feedstock/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    - rust
+    - libzlib
+    - openssl
+    - cyrus-sasl
+    - perl
+    - python
+    - pip
+    - protobuf
+    - librdkafka
+
```
Comment (ocefpaf) [inline recipes/bytewax-feedstock/meta.yaml:40]:
```suggestion
```

File: recipes/bytewax-feedstock/meta.yaml (hunk 1 above)
Comment (cav71) [reply recipes/bytewax-feedstock/meta.yaml:14]: done

**Comment by cav71** [comment]
@zklaus @ocefpaf I've placed a request to the mainstream package [bytewax](https://github.com/bytewax/bytewax/issues/265). The feedstock LGTM but I'd be waiting if the upstream maintainers have willingness/time to collaborate on this.

File: recipes/bytewax-feedstock/bld.bat [hunk 5]
```diff
@@ -0,0 +1,7 @@
+@echo on
+SET RUST_BACKTRACE=1
+SET OPENSSL_DIR=%PREFIX%
+SET OPENSSL_LIB_DIR=%PREFIX%\lib
+SET OPENSSL_INCLUDE_DIR=%PREFIX%\include
+%PYTHON% -m pip install . -vv
+
```
Comment (ocefpaf) [inline recipes/bytewax-feedstock/bld.bat:7]: This file is not used b/c you are skipping Windows. Do you want to keep it and try again in the feedstock or do you want to remove it?

File: recipes/bytewax-feedstock/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - curl
+    - {{ compiler('cxx') }}
+    - make
+    - pkg-config
+    - python
```
Comment (ocefpaf) [inline recipes/bytewax-feedstock/meta.yaml:27]: In theory you should only need `{{ compiler('cxx') }}` in build. Sometimes, when cross-compiling, you may need Python and/or others. Can you try to remove these and/or move to `host` to make sure you need them here?

File: recipes/bytewax-feedstock/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  host:
+    - cmake
+    - maturin
+    - rust
```
Comment (ocefpaf) [inline recipes/bytewax-feedstock/meta.yaml:31]: If you need the compiler you need to declare it like in https://github.com/conda-forge/cryptography-feedstock/blob/32ac6dabfba953884ee338e40e183fb2d1ed6ac2/recipe/meta.yaml#L67

```suggestion
    - rust
```

File: recipes/bytewax-feedstock/bld.bat (hunk 5 above)
Comment (cav71) [reply recipes/bytewax-feedstock/bld.bat:7]: The windows build fails, but I cannot figure out the reason from the logs: I was trying to see if I can get hold of a win machine in order to debug the build.

Comment (ocefpaf) [reply recipes/bytewax-feedstock/bld.bat:7]: You can skip it if the solution is too complex and open an issue in the feedstock to fix it later.

File: recipes/bytewax-feedstock/meta.yaml (hunk 6 above)
Comment (cav71) [reply recipes/bytewax-feedstock/meta.yaml]: done

**Comment by ocefpaf** [comment]
> Alas, I don't have permissions and thus am not one of them.

Be careful with what you wish for ;-p
Jokes aside, you can request to be a formal staged-recipes reviewer and get commit rights here.

@cav71 is this one ready to go?

**Comment by cav71** [comment]
Yes, it passes the tests. 
Talking of whishes, I'd wish I had access to a winbox to build/test the packaging there, but linux/osx are fine.
Thanks

**Comment by ocefpaf** [comment]
> I'd wish I had access to a winbox to build/test the packaging there, but linux/osx are fine.

You can use the CIs here or, better yet, in the feedstock once we merge this. Sadly conda-forge has zero "own" infrastructure. We rely on CIs.

**Pull request merged**